external/llvm-project/mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -411,6 +412,47 @@ def AMDGPU_RawBufferAtomicUminOp :
   let hasVerifier = 1;
 }
 
+def AMDGPU_DPP : AMDGPU_Op<"dpp", [SameTypeOperands, AllTypesMatch<["result", "src"]>]>,
+  Arguments<(ins AnyType:$src,
```

Comment:
While we're here, since I started looking at things the compiler itself does and recently read the public example we have of how to do a DPP reduction:
- Add an optional operand called  `old` . If `old` is present, lower to `update.dpp` instead of `mov.dpp`.
- (which means you'll need to crack the rocdl files open and provide a wrapper around that intrinsic and add the relevant tests, but that should be mostly a copy-paste job once the definition of mov.dpp's been fixed per below)